src/tck/Reactive.Streams.TCK/Support/AtomicCounterLong.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System.Threading;
+
+namespace Reactive.Streams.TCK.Support
+{
+    /// <summary>
+    /// An atomic 64 bit integer counter.
+    /// </summary>
+    public class AtomicCounterLong
+    {
+        /// <summary>
+        /// Creates an instance of an AtomicCounterLong.
+        /// </summary>
+        /// <param name="value">The initial value of this counter.</param>
+        public AtomicCounterLong(long value)
+        {
+            _value = value;
+        }
+
+        /// <summary>
+        /// Creates an instance of an AtomicCounterLong with a starting value of -1.
+        /// </summary>
+        public AtomicCounterLong()
+        {
+            _value = -1;
+        }
+
+        /// <summary>
+        /// The current value for this counter.
+        /// </summary>
+        private long _value;
+
+        /// <summary>
+        /// Retrieves the current value of the counter
+        /// </summary>
+        public long Current => Interlocked.Read(ref _value);
```

Comment:
Interesting to see how that's handled on CLR. Name implies a lock is involved, however docs seem to explain that it's "just one atomic op" - is this like atomic ops on the jvm or more like synchronizing (serializing access)? (Just for my reference, impl is good in any case :-))